odysseus/odysseus_tree/board/tpu/plugin_user_property_timestamp.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+
+#include "nanomq/plugin.h"
+
+/*
+ * How to compile:
+ * gcc -I../ -fPIC -shared plugin_user_property.c -o plugin_user_property.so
+ */
+
+
+int cb(void *data)
+{
+	char **property = data;
+	if (property != NULL) {
+		
+		// key of ts, add 1 bc termination byte
+		property[0] = malloc(strlen("ts") + 1);
+		strcpy(property[0], "ts");
+		
+		struct timeval tv;
+  
+		gettimeofday(&tv, NULL);
+  
+		unsigned long long millisecondsSinceEpoch =
+			(unsigned long long)(tv.tv_sec) * 1000 +
+			(unsigned long long)(tv.tv_usec) / 1000;
+			
+		// currently the time is using 13 characters, so 30 should be enough
+ 		char str[30]; 
+ 		// convert the epoch time to a char arr
+		sprintf(str, "%llu", millisecondsSinceEpoch);
+// 		
+// 		// value of the time, add 1 bc termination byte
+ 		property[1] = malloc(strlen(str)+1);
```

Comment:
same thing here i think you can use sprintf

Reply:
see above

Reply:
I agree with Peyton this seems like a memory leak source

Reply:
see above again lol